src/OrleansServiceBus/Providers/Streams/EventHub/EventHubQueueCacheFactory.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Orleans.Providers.Streams.Common;
+using Orleans.Runtime;
+using Orleans.Serialization;
+using Orleans.Streams;
+
+namespace Orleans.ServiceBus.Providers
+{
+    public class EventHubQueueCacheFactory : IEventHubQueueCacheFactory
+    {
+        private readonly EventHubStreamProviderSettings _providerSettings;
+        private readonly SerializationManager _serializationManager;
+        private readonly IObjectPool<FixedSizeBuffer> _bufferPool;
```

Comment:
This pattern assumes that the same buffer pool is used per cache, which is the default, but not the only supported pattern.
We support a buffer pool:
- Per silo, which can be done here by passing the same buffer pool to each cache factory for all providers.
- Per stream provider, which is the default, and what this implementation does by default.
- Per cache, which this does not support.

I'm not necessarily suggesting you change this, as this is a default utility class, and to that end, it does provide utility, I'm only raising the point that this class will not be useful to those who need a block pool per queue, which is not an uncommon pattern.